pkg/controller/clusterconnection/clusterconnection_controller.go

```diff
@@ -349,7 +352,7 @@ func (r *ReconcileConnection) Reconcile(ctx context.Context, request reconcile.R
 	// In managed clusters, the clusterconnection controller is a dependency for the License to be created. In case the
 	// License is unavailable and reconciliation of non-NetworkPolicy resources in the clusterconnection controller
 	// would resolve it, we render network policies last to prevent a chicken-and-egg scenario.
-	if includeV3NetworkPolicy {
+	if includeV3NetworkPolicy && isCalicoEnterprise {
```

Comment:
we shouldn't need this extra check here - `includeV3NetworkPolicy` will be false if `isCalicoEnterprise` is false.